src/v/redpanda/admin/api-doc/debug_bundle.def.json

```diff
@@ -0,0 +1,123 @@
+"scram_authn": {
+  "type": "object",
+  "required": ["mechanism", "username", "password"],
+  "properties": {
+    "mechanism": {
+      "description": "SCRAM mechanism",
+      "type": "string"
+    },
+    "username": {
+      "description": "username used by RPK to authenticate against Kafka and Admin API",
+      "type": "string"
+    },
+    "password": {
+      "description": "password used by RPK to authenticate against Kafka and Admin API",
+      "type": "string"
+    }
+  }
+},
+"bundle_start_config_params": {
+  "type": "object",
+  "required": [],
+  "properties": {
+    "authentication": {
+      "description": "Authentication object",
+      "type": "scram_authn"
+    },
+    "controller_logs_size_limit_bytes": {
+      "type": "int"
+    },
+    "cpu_profiler_wait_seconds": {
+      "type": "int"
+    },
+    "logs_since": {
+      "description": "Include logs dated from specified date onward; (journalctl date format: YYYY-MM-DD, 'yesterday', or 'today'). Refer to journalctl documentation for more options.",
+      "type": "string"
+    },
+    "logs_size_limit_bytes": {
+      "type": "int"
+    },
+    "logs_until": {
+      "description": "Include logs older than the specified date; (journalctl date format: YYYY-MM-DD, 'yesterday', or 'today'). Refer to journalctl documentation for more options.",
+      "type": "string"
+    },
+    "metrics_interval_seconds": {
+      "type": "int"
+    },
+    "partition": {
+      "type": "string"
+    }
+  }
+},
+"bundle_start_config": {
+  "type": "object",
+  "required": ["job_id"],
+  "properties": {
+    "job_id": {
+      "description": "The Job UUID to use",
+      "type": "string"
+    },
+    "config": {
+      "description": "The parameters",
+      "type": "bundle_start_config_params"
+    }
+  }
+},
+"bundle_start_response": {
+  "type": "object",
+  "required": ["job_id"],
+  "properties": {
+    "job_id": {
+      "description": "The Job UUID associated with the debug bundle process",
+      "type": "string"
+    }
+  }
+},
+"get_bundle_status": {
+  "type": "object",
+  "required": ["job_id", "status", "created"],
+  "properties": {
+    "job_id": {
+      "description": "The Job UUID associated with the debug bundle process",
+      "type": "string"
+    },
+    "status": {
+      "description": "The status of the job",
+      "type": "string"
+    },
+    "created": {
+      "description": "When the job was started, in milliseconds since epoch",
+      "type": "int"
+    },
+    "filename": {
```

Comment:
why do we need a file name?  shouldn't the file just be named job_id.tgz?

Reply:
> why do we need a file name? shouldn't the file just be named job_id.tgz?

Since the UUID is user-supplied, I guess it's possible for the file to already exist, in which case we'd have to create a different filename.

I think this may have been brought up in the RFC.

Reply:
I did read the RFC but I guess I missed this there.  Why do we let the user supply a UUID?  It seems better to automate this, may need to check with Mike on the rationale here.